[IntelliJ][ReportedByUser] Uncaught Exception  java.lang.Throwable: Control-flow exceptions (like ProcessCanceledException) should never be logged

IntelliJ build version: 2020.2 IU-202.6397.94
OS: Windows 10
JDK: JetBrains s.r.o. 11.0.7
Plugin version: 3.40.0-2020.2
Additional Info: None
Parent component: 
```
com.intellij.openapi.wm.impl.IdeRootPane[,7,7,1505x963,layout=com.intellij.openapi.wm.impl.IdeRootPane$MyRootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=449,maximumSize=,minimumSize=,preferredSize=]
```
Error message: 
```
IdeaLoggingEvent[message=null, throwable=java.lang.Throwable: Control-flow exceptions (like ProcessCanceledException) should never be logged
	at com.intellij.openapi.diagnostic.Logger.checkException(Logger.java:195)
	at com.microsoft.intellij.AzurePlugin.log(AzurePlugin.java:406)
	at com.microsoft.intellij.AzurePlugin.initializeAIRegistry(AzurePlugin.java:249)
	at com.microsoft.intellij.AzurePlugin.runActivity(AzurePlugin.java:117)
Caused by: com.intellij.openapi.progress.ProcessCanceledException
	at com.microsoft.intellij.AzureSettings.getSafeInstance(AzureSettings.java:51)
	at com.microsoft.intellij.AzurePlugin.initializeAIRegistry(AzurePlugin.java:225)
```

--- a/PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/AzurePlugin.java
+++ b/PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/AzurePlugin.java
@@ -38,6 +38,7 @@ import com.intellij.openapi.progress.ProcessCanceledException;
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.startup.StartupActivity;
 import com.intellij.openapi.startup.StartupManager;
+import com.intellij.util.ExceptionUtil;
 import com.intellij.util.PlatformUtils;
 import com.intellij.util.containers.HashSet;
 import com.microsoft.applicationinsights.preference.ApplicationInsightsResource;
@@ -71,6 +72,7 @@ import java.io.*;
 import java.net.URL;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 import java.util.Set;
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipInputStream;
@@ -259,7 +261,11 @@ public class AzurePlugin implements StartupActivity.DumbAware {
             }
             AzureSettings.getSafeInstance(myProject).saveAppInsights();
         } catch (Exception ex) {
-            AzurePlugin.log(ex.getMessage(), ex);
+            // https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000093184-What-is-com-intellij-openapi-progress-ProcessCanceledException
+            // should ignore ProcessCanceledException
+            if (Objects.isNull(ExceptionUtil.findCause(ex, ProcessCanceledException.class))) {
+                AzurePlugin.log(ex.getMessage(), ex);
+            }
         }
     }
 

